django/core/paginator.py

```diff
@@ -52,6 +53,9 @@ def page(self, number):
         top = bottom + self.per_page
         if top + self.orphans >= self.count:
             top = self.count
+
+        if hasattr(self.object_list, 'ordered') and not self.object_list.ordered:
+            warnings.warn("Inconsistent pagination when no ordering is specified", RuntimeWarning)
```

Comment:
Alright, let's use `RuntimeWarning` in this case and I'll post to @django-developers about it.